ldk/javascript/examples/self-test-loop/src/tests/whisper/index.ts

```diff
@@ -244,6 +245,93 @@ export const testBoxInBox = (): Promise<boolean> =>
     }
   });
 
+export const testDropzone = async (): Promise<boolean> => {
+  const dropZone: whisper.DropZone = {
+    type: WhisperComponentType.DropZone,
+    onDrop: () => {},
+    label: 'File Components',
+    key: 'drop',
+  };
+  const onDropAction = new Promise<whisper.File[]>((resolve, reject) => {
+    dropZone.onDrop = (error, param) => {
+      if (error) {
+        reject(error);
+      }
+      if (param) {
+        resolve(param);
+      }
+    };
+  });
+  const testWhisper = await whisper.create({
+    label: 'Dropzone Test',
+    components: [
+      {
+        type: WhisperComponentType.Markdown,
+        body: 'Select and drop a file onto the dropzone box below.',
+      },
+      dropZone,
+    ],
+  });
+  const droppedFiles = await onDropAction;
+  const fileData = droppedFiles
+    .map((file) => `Path: ${file.path}, Size: ${file.size}`)
+    .join('\n\n');
+
+  function createAcceptButtons(): {
```

Comment:
There are already 'resolveRejectButtons' in the whisper/utils which I believe would work for you here.

Reply:
Thanks :) I'll take a look!

Reply:
Implemented! Though I kept the function b/c I like getting the promise returned, as opposed to supplying my own.